src/statusBar/pollStatusBar.ts

```diff
@@ -3,21 +3,25 @@ import { getStatus } from "../binary/requests/statusBar";
 import { BINARY_STATUS_BAR_FIRST_MESSAGE_POLLING_INTERVAL } from "../globals/consts";
 import {
   onStartServiceLevel,
-  pollServiceLevel,
   resetDefaultStatus,
+  setServiceLevel,
 } from "./statusBar";
 import handleStatus, {
   disposeStatusBarCommand,
 } from "./statusBarActionHandler";
+import { statePoller } from "../state/statePoller";
 
 let statusPollingInterval: NodeJS.Timeout | null = null;
 
 export default function pollStatuses(
   context: vscode.ExtensionContext
 ): vscode.Disposable {
+  const statePollerDisposable = statePoller.event((change) => {
```

Comment:
1 - what problem does this solves?
2 - why do we need the both pollings here  ? ie statePoller and doPollStatus ?

Reply:
they are emitting at different intervals

Reply:
state poller is state of user

status poller is status bar messages from the binary

Reply:
ok